### ISSUE TITLE
Linux aarch64 SDK support

### ISSUE THREAD

**Issue by koroshiya** [kobaltcore/renkit#22]
Note: This is not the same as issue #19 
This suggestion is about adding aarch64 as a target, rather than a supported platform.

As noted in that issue, there is an aarch64 version of Renpy.
What's worth noting, and why I've raised this issue, is that the aarch64 build isn't only useful when building on aarch64 platforms.

The aarch64 build contains both x86 and aarch64 libraries.
So it will run from an x86_64 machine, yet output Linux builds which include both the aarch64 and x86 libraries.
You can then distribute builds of a game to people on those platforms (Raspberry Pi, Chromebook) while still using an x86 machine and x86 build of renkit and renpy.

Currently renkit downloads a .zip file from dl/{self}/renpy-{self}-sdk.zip
The aarch64 version is hosted on dl/{self}/renpy-{self}-sdkarm.tar.bz2

The path and extension are different, so using the aarch64 version isn't as simple as passing renkit a different version number.
renkit would need to be updated to use the different url format for aarch64, add bzip2 and tar to Cargo, and extract with those instead of the zip utilities in order for this to work.

I'm not suggesting that this would be the default. Just that it would be an option.
eg. renkit could allow you to install 7.7.3-arm instead of 7.7.3

For my own use case I've instead modified the Dockerfile, since I was going to run it in a container anyway.

```
# build for a specific version of Ren'Py with:
# docker build . --tag renpy:8.2.0 --build-arg renpy_version=8.2.0
# to run commands:
# docker run --rm -it --volume /local/project/path:/project renpy:8.2.0 renutil launch 8.2.0 -d -- "/project compile"

FROM ubuntu:22.04
ARG renpy_version=8.2.0
ARG renkit_version=v4.3.0

ENV DEBIAN_FRONTEND=noninteractive

# install java
ENV JAVA_HOME=/opt/java/openjdk
COPY --from=eclipse-temurin:21-jdk $JAVA_HOME $JAVA_HOME
ENV PATH="${JAVA_HOME}/bin:${PATH}"

# install dependencies
RUN apt-get update && \
    apt-get install -y curl wget xz-utils libgl1 bzip2 && \
    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*

ENV PATH="/root/.cargo/bin:${PATH}"

# install renkit
RUN curl --proto '=https' --tlsv1.2 -LsSf https://github.com/kobaltcore/renkit/releases/download/$renkit_version/renkit-installer.sh | sh

# install the specified version of Ren'Py
RUN $HOME/.cargo/bin/renutil install $renpy_version

# download the aarch64 version of renpy
RUN wget https://www.renpy.org/dl/$renpy_version/renpy-$renpy_version-sdkarm.tar.bz2 

# extract and delete the archive
RUN tar xf renpy-$renpy_version-sdkarm.tar.bz2
RUN rm renpy-$renpy_version-sdkarm.tar.bz2

# move rapt from the x86 version to the aarch64 version
RUN mv $HOME/.renutil/$renpy_version/rapt renpy-$renpy_version-sdkarm

# delete the x86 version
RUN rm -rf $HOME/.renutil/$renpy_version

# move and rename the aarch64 version
RUN mv renpy-$renpy_version-sdkarm $HOME/.renutil/$renpy_version

# default entrypoint so people can dispatch to any of renkit's tools
CMD [ "/bin/bash", "-c"]
```

The above added one new package to install (bzip2).
The other changes are between "renutil install" and CMD.
It's a crude workaround, but it seems to do what I need it to.

**Comment by kobaltcore** [comment]
Yeah, that's a good call. I was always hesitant because of the special status the ARM version seems to have, even on the official download page. I have since talked to `renpytom` and he says this is purely to save on file size for libs that most people won't need. Whatever the reason may be, it seems that this is unlikely to change unless ARM Linux gets a big user boost, so the current way of things is decently stable.

I'll make a note to implement support for choosing the ARM version whenever it exists, to allow users to deploy to as many targets as possible 👍

**Comment by kobaltcore** [comment]
I've made a first tentative implementation of this. Having to pull in a `bzip2` dependency is a bit unfortunate, but alas the the pure-Rust implementation of `bzip2` currently doesn't work for this use case, so there's no better option.

The change is internal-only: `renutil` will simply pull the version with ARM support if it's available, otherwise it'll fall back to the 'normal' variant so nothing changes in the frontend.

Due to the new dependency on a system library some more testing is needed before a full release, which is why I'm pushing an alpha version of `4.4.0` for now, which I'll work on until I can be reasonably sure that everything still behaves properly on varying systems.

Feel free to give that version a try yourself if you wanna help in the endeavor, it's [available](https://github.com/kobaltcore/renkit/releases/tag/v4.4.0-alpha.1) right now.

**Comment by koroshiya** [comment]
Thanks kobaltcore, I've just given this a test in a new docker container.
I've used the Dockerfile from this repository as-is, but with renkit_version=v4.4.0-alpha.1 and renpy_version=7.7.3
All worked as intended.

Entering the container with
`docker run  -it renpy:7.7.3-test bash`
I can see the ARM libs in /root/.renutil/7.7.3/lib/

After compiling a linux build of a game and extracting the archive I can likewise see the arm libs in the compiled game.
I don't have an arm device setup to test with right now, but looking at the output it all looks good, as far as I can tell.

**Comment by kobaltcore** [comment]
Thanks, that's good news. I'm running on an M1 mac myself so I was able to test this on an `aarch64` machine, where I promptly discovered that I forgot to add handling for these extra architectures in my `Version` implementation. This should now be fixed, although I have no way of testing this on a 32-bit ARM processor, as I don't have something like a Raspberry Pi available at the moment.
Still, the new `alpha.2` release should work a bit better now. I'll do a bit more testing before I promote to a full release though.

**Comment by kobaltcore** [comment]
I wasn't able to turn up any further issues in `alpha.2` so I've gone ahead and released `v4.4.0`. I'll close this issue as completed.
In case something else pops up, feel free to make a new issue and I'll look at it as soon as I'm able.
